src/mesh/p4est_mesh.jl

```diff
@@ -0,0 +1,399 @@
+      tree_to_face[1, tree] = 1
+    elseif periodicity[1]
+      tree_to_tree[1, tree] = linear_indices[cells_x, cell_y] - 1
+      tree_to_face[1, tree] = 1
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[1, tree] = tree - 1
+      tree_to_face[1, tree] = 0
+    end
+
+    # Positive x-direction
+    if cell_x < cells_x
+      tree_to_tree[2, tree] = linear_indices[cell_x + 1, cell_y] - 1
+      tree_to_face[2, tree] = 0
+    elseif periodicity[1]
+      tree_to_tree[2, tree] = linear_indices[1, cell_y] - 1
+      tree_to_face[2, tree] = 0
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[2, tree] = tree - 1
+      tree_to_face[2, tree] = 1
+    end
+
+    # Negative x-direction
+    if cell_y > 1
+      tree_to_tree[3, tree] = linear_indices[cell_x, cell_y - 1] - 1
+      tree_to_face[3, tree] = 3
+    elseif periodicity[2]
+      tree_to_tree[3, tree] = linear_indices[cell_x, cells_y] - 1
+      tree_to_face[3, tree] = 3
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[3, tree] = tree - 1
+      tree_to_face[3, tree] = 2
+    end
+
+    # Positive x-direction
+    if cell_y < cells_y
+      tree_to_tree[4, tree] = linear_indices[cell_x, cell_y + 1] - 1
+      tree_to_face[4, tree] = 2
+    elseif periodicity[2]
+      tree_to_tree[4, tree] = linear_indices[cell_x, 1] - 1
+      tree_to_face[4, tree] = 2
+    else # Non-periodic boundary, tree and face point to themselves (zero-based indexing)
+      tree_to_tree[4, tree] = tree - 1
+      tree_to_face[4, tree] = 3
+    end
+  end
+
+  tree_to_corner = C_NULL
+  # p4est docs: "in trivial cases it is just a pointer to a p4est_topix value of 0."
+  # We don't need corner connectivity, so this is a trivial case.
+  ctt_offset = Array{p4est_topidx_t}([0])
+
+  corner_to_tree = C_NULL
+  corner_to_corner = C_NULL
+
+  p4est_connectivity_new_copy(n_vertices, n_trees, n_corners,
+                              vertices, tree_to_vertex,
+                              tree_to_tree, tree_to_face,
+                              tree_to_corner, ctt_offset,
+                              corner_to_tree, corner_to_corner)
+end
+
+@inline Base.ndims(::P4estMesh{NDIMS}) where NDIMS = NDIMS
+@inline Base.real(::P4estMesh{NDIMS, RealT}) where {NDIMS, RealT} = RealT
+
+@inline ntrees(mesh::P4estMesh) = mesh.p4est.trees.elem_count
+@inline ncells(mesh::P4estMesh) = mesh.p4est.global_num_quadrants
+
+
+function Base.show(io::IO, mesh::P4estMesh)
+  print(io, "P4estMesh{", ndims(mesh), ", ", real(mesh), "}")
+end
+
+
+function Base.show(io::IO, ::MIME"text/plain", mesh::P4estMesh)
+  if get(io, :compact, false)
+    show(io, mesh)
+  else
+    setup = [
+             "#trees" => ntrees(mesh),
+             "current #cells" => ncells(mesh),
+             "polydeg" => length(mesh.nodes),
```

Comment:
```suggestion
             "#nodes" => length(mesh.nodes),
```

Reply:
I changed it to `polydeg => length(mesh.nodes) - 1` somewhere. Why do you prefer `#nodes`?

Reply:
I guess I changed it in my 3D branch, hoping that no one notices before.

Reply:
`polydeg` is very specific to DGSEM while `nnodes` is more general and includes other DG-type methods such as the finite difference schemes implemented in #617

Reply:
This is independent of the solver nodes, though. It just defines the polynomial degree of the approximation of the geometry (which, of course, can't be higher than the polydeg of the DGSEM solver for non-conforming meshes).

Reply:
Hmm, okay.